### DISCUSSION TITLE
How many points/graphs can GNN's handle?

### DISCUSSION THREAD

**Discussion by davodogster** [pyg-team/pytorch_geometric#3950] [General]
Hi there, I have a pointcloud dataset of objects with object-level annotations, each object has between 10,000 and 200,000 points. Is it possible to train a GNN with samples this large? I understand that the point cloud files need to be converted to graphs first?

IIF it was a Point Cloud Neural network I would have to voxelize the point clouds, or I would have to downsample them all to e.g 2048 points which is far too small for my task.

I want to train a GNN to learn the geometric features. Could you provide any pointers on how to go about this?

**Comment by rusty1s** [comment]
I think you may be able to handle up to 50k-100k points in one batch, but the actual memory consumption does depend more on the number of edges rather than the number of nodes. As a result, you can either try to split point clouds with a large number of nodes into multiple examples, or do some farthest point sampling via `torch_cluster.fps` to reduce the initial size by a certain factor.
